include/libembeddedhal/math.hpp

```diff
@@ -95,4 +95,30 @@ template<typename T>
 
   return quotient;
 }
+
+/**
+ * @brief Remap a value from one range to another without constraining values
+ * within the given ranges.
+ *
+ * @tparam T
+ * @param p_in_value - Value to convert.
+ * @param p_in_min - Minimum value of the range to convert from.
+ * @param p_in_max - Maximum value of the range to convert from.
+ * @param p_out_min - Minimum value of the range to convert to.
+ * @param p_out_max - Maximum value of the range to convert to.
+ * @return constexpr T - Value after being remapped to out range
+ */
+template<typename T>
+[[nodiscard]] constexpr T linear_map(T p_in_value,
```

Comment:
```suggestion
[[nodiscard]] constexpr boost::leaf::result<T> linear_map(T p_in_value,
```

Should be able to return an error.